File: backend/models/company.py
"""Company entity (docs/V2/DATA_MODEL.md).

Represents an organization monitored by Scout. Exists independently of
any research - Research Sessions, Opportunities, and Reports reference a
Company but a Company can exist with none yet.
"""

import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class Company(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    name: str
    industry: Optional[str] = None
    headquarters: Optional[str] = None
    website: Optional[str] = None
    # String rather than a bool/enum - DATA_MODEL.md names this attribute
    # "Monitoring Status" (not "Monitoring Enabled"); Phase 3 (Company
    # Management) is where enable/disable behavior actually gets wired in.
    monitoring_status: str = "enabled"
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/react/src/types/company.ts
export interface Company {
  id: string;
  name: string;
  industry: string | null;
  headquarters: string | null;
  website: string | null;
  monitoring_status: string;
  // Priority 5 (soft delete/archive) - null means active; set means
  // archived. Orthogonal to monitoring_status.
  archived_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateCompanyInput {
  name: string;
  industry?: string;
  headquarters?: string;
  website?: string;
}
